From 3c771fc6111d2703274d269b2b34183ce8be4df4 Mon Sep 17 00:00:00 2001
From: Diana Craciun <diana.craciun@nxp.com>
Date: Mon, 18 Nov 2019 11:37:59 +0200
Subject: [PATCH] bus/fsl-mc: Set the QMAN/BMAN region flags

The QMAN region can be memory mapped, so it should be
of type IORESOURCE_MEM. Also use the bus specific bits
in order to pass additional information about the region.

Signed-off-by: Diana Craciun <diana.craciun@nxp.com>
---
 drivers/bus/fsl-mc/fsl-mc-bus.c | 7 ++-----
 1 file changed, 2 insertions(+), 5 deletions(-)

--- a/drivers/bus/fsl-mc/fsl-mc-bus.c
+++ b/drivers/bus/fsl-mc/fsl-mc-bus.c
@@ -627,11 +627,8 @@ static int fsl_mc_device_get_mmio_region
 
 		regions[i].end = regions[i].start + region_desc.size - 1;
 		regions[i].name = "fsl-mc object MMIO region";
-		regions[i].flags = IORESOURCE_IO;
-		if (region_desc.flags & DPRC_REGION_CACHEABLE)
-			regions[i].flags |= IORESOURCE_CACHEABLE;
-		if (region_desc.flags & DPRC_REGION_SHAREABLE)
-			regions[i].flags |= IORESOURCE_MEM;
+		regions[i].flags = region_desc.flags & IORESOURCE_BITS;
+		regions[i].flags |= IORESOURCE_MEM;
 	}
 
 	mc_dev->regions = regions;
